samza-log4j/src/main/java/org/apache/samza/logging/log4j/StreamAppender.java

```diff
@@ -93,6 +97,17 @@ public void setStreamName(String streamName) {
     this.streamName = streamName;
   }
 
+  public int getDefaultPartitionCount() {
+    if (defaultPartitionCount > 0) {
+      return defaultPartitionCount;
+    }
+    return new JobConfig(getConfig()).getContainerCount();
+  }
+
+  public void setDefaultPartitionCount(int defaultPartitionCount) {
```

Comment:
Can you add javadoc to this method to explain up to what point in the container lifecycle may this be called? In other words, when is activateOptions called?